snapcraft/internal/parts.py

```diff
@@ -270,7 +270,8 @@ def load_plugin(self, part_name, plugin_name, part_properties):
             plugin_name=plugin_name,
             part_properties=part_properties,
             project_options=self._project_options,
-            part_schema=self._validator.part_schema)
+            part_schema=self._validator.part_schema,
+            definitions_schema=self._validator.definitions_schema)
```

Comment:
Yeah definitely. I didn't actually realize those replacements could be recursive (I see that now). Want me to try that? Or would we rather move to all json in the future, in which case this would need to stay?